### PR TITLE
sweeper/aws_rds_global_cluster: Fixes dependency errors

### LINKED ISSUE #26234: resource/aws_db_instance: Sweeper fails on last instance of master cluster
### Description

> error deleting DB Instance (...): InvalidDBClusterStateFault: Cannot delete the last instance of the master cluster. Delete the replica cluster before deleting the last master cluster instance.

### Affected Resource

* aws_db_instance

### LINKED ISSUE #26236: resource/aws_rds_cluster: Cannot sweep primary member of global cluster
### Description

If an `aws_rds_cluster` has secondary global cluster members, the primary fails to delete with 

> Failed to delete RDS DB Cluster (...): InvalidGlobalClusterStateFault: Unable to delete cluster that is a primary member of global cluster while there are other global cluster members

### Affected Resource

* aws_rds_cluster

### PR DESCRIPTION
**Pull request by gdavison**
Currently, deleting an RDS Global Cluster can fail with either

> error deleting DB Instance (...): InvalidDBClusterStateFault: Cannot delete the last instance of the master cluster. Delete the replica cluster before deleting the last master cluster instance.

or

> Failed to delete RDS DB Cluster (...): InvalidGlobalClusterStateFault: Unable to delete cluster that is a primary member of global cluster while there are other global cluster members

This PR updates the RDS Global Cluster Delete operation to remove all member clusters from the global cluster.

### Relations

Closes #26234
Closes #26236

### Output from Acceptance Testing

```console
$ make sweep SWEEPARGS=-sweep-run=aws_rds_global_cluster

2023/02/16 17:45:37 [DEBUG] Running Sweepers for region (us-west-2):
2023/02/16 17:45:37 [DEBUG] Running Sweeper (aws_rds_global_cluster) in region (us-west-2)
2023/02/16 17:45:37 [DEBUG] Completed Sweeper (aws_rds_global_cluster) in region (us-west-2) in 493.334833ms
2023/02/16 17:45:37 [DEBUG] Sweeper (aws_rds_cluster_parameter_group) has dependency (aws_rds_cluster), running..
2023/02/16 17:45:37 [DEBUG] Sweeper (aws_rds_cluster) has dependency (aws_db_instance), running..
2023/02/16 17:45:37 [DEBUG] Sweeper (aws_db_instance) has dependency (aws_rds_global_cluster), running..
2023/02/16 17:45:37 [DEBUG] Sweeper (aws_rds_global_cluster) already ran in region (us-west-2)
2023/02/16 17:45:37 [DEBUG] Running Sweeper (aws_db_instance) in region (us-west-2)
2023/02/16 17:45:37 [DEBUG] Completed Sweeper (aws_db_instance) in region (us-west-2) in 48.134125ms
2023/02/16 17:45:37 [DEBUG] Running Sweeper (aws_rds_cluster) in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Completed Sweeper (aws_rds_cluster) in region (us-west-2) in 45.59775ms
2023/02/16 17:45:38 [DEBUG] Running Sweeper (aws_rds_cluster_parameter_group) in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Completed Sweeper (aws_rds_cluster_parameter_group) in region (us-west-2) in 56.894875ms
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_rds_cluster) has dependency (aws_db_instance), running..
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_db_instance) has dependency (aws_rds_global_cluster), running..
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_rds_global_cluster) already ran in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_db_instance) already ran in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_rds_cluster) already ran in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_db_instance) has dependency (aws_rds_global_cluster), running..
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_rds_global_cluster) already ran in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_db_instance) already ran in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_db_instance_automated_backups_replication) has dependency (aws_db_instance), running..
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_db_instance) has dependency (aws_rds_global_cluster), running..
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_rds_global_cluster) already ran in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Sweeper (aws_db_instance) already ran in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Running Sweeper (aws_db_instance_automated_backups_replication) in region (us-west-2)
2023/02/16 17:45:38 [DEBUG] Waiting for state to become: [success]
2023/02/16 17:45:38 [DEBUG] Waiting for state to become: [success]
2023/02/16 17:45:38 [DEBUG] Waiting for state to become: [success]
2023/02/16 17:45:38 [DEBUG] Waiting for state to become: [success]
2023/02/16 17:45:38 [DEBUG] Waiting for state to become: [success]
2023/02/16 17:45:38 [DEBUG] Waiting for state to become: [success]
2023/02/16 17:45:38 [DEBUG] Waiting for state to become: [success]
2023/02/16 17:45:38 [DEBUG] Stopping RDS Instance Automated Backups Replication: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-srqgk6fdu6hghmeciso2xfyzat2akho3aqu632y
2023/02/16 17:45:38 [DEBUG] Stopping RDS Instance Automated Backups Replication: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-dkedyvwfa2niqxut6p23q3yaz2jli2utyoobf3q
2023/02/16 17:45:38 [DEBUG] Stopping RDS Instance Automated Backups Replication: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-pxokasoxjknalh3xd6qqyve7zcqxz2m6txij7iq
2023/02/16 17:45:38 [DEBUG] Stopping RDS Instance Automated Backups Replication: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-rvdjvrrsiydqdespjwrlitkiqdf5yxhuoraqnky
2023/02/16 17:45:38 [DEBUG] Stopping RDS Instance Automated Backups Replication: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-7i4nrfgq2mhatsqzhoe55xpvulboatceapoqp6y
2023/02/16 17:45:38 [DEBUG] Stopping RDS Instance Automated Backups Replication: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-dikxdana362y7pmwlq2mku7gyyhwj4bbp5vducq
2023/02/16 17:45:38 [DEBUG] Stopping RDS Instance Automated Backups Replication: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-25xuy5dumde6sme4x7x3sjldw64cgoaacsusbyi
2023/02/16 17:45:38 [DEBUG] Deleting RDS Instance Automated Backup: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-25xuy5dumde6sme4x7x3sjldw64cgoaacsusbyi
2023/02/16 17:45:39 [ERROR] deleting RDS Instance Automated Backup (arn:aws:rds:us-west-2:123456789012:auto-backup:ab-25xuy5dumde6sme4x7x3sjldw64cgoaacsusbyi): InvalidDBInstanceAutomatedBackupState: You can't delete automated backups.
	status code: 400, request id: 84faaf79-6557-495f-bc0e-a4d37fd73b9a
2023/02/16 17:45:39 [DEBUG] Deleting RDS Instance Automated Backup: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-7i4nrfgq2mhatsqzhoe55xpvulboatceapoqp6y
2023/02/16 17:45:39 [ERROR] deleting RDS Instance Automated Backup (arn:aws:rds:us-west-2:123456789012:auto-backup:ab-7i4nrfgq2mhatsqzhoe55xpvulboatceapoqp6y): InvalidDBInstanceAutomatedBackupState: You can't delete automated backups.
	status code: 400, request id: 84ae31a8-6e96-4c80-a44e-e57efea20eee
2023/02/16 17:45:39 [DEBUG] Deleting RDS Instance Automated Backup: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-dikxdana362y7pmwlq2mku7gyyhwj4bbp5vducq
2023/02/16 17:45:39 [ERROR] deleting RDS Instance Automated Backup (arn:aws:rds:us-west-2:123456789012:auto-backup:ab-dikxdana362y7pmwlq2mku7gyyhwj4bbp5vducq): InvalidDBInstanceAutomatedBackupState: You can't delete automated backups.
	status code: 400, request id: 0b323345-04dd-40b7-a53b-658cc2f49e35
2023/02/16 17:45:39 [DEBUG] Deleting RDS Instance Automated Backup: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-dkedyvwfa2niqxut6p23q3yaz2jli2utyoobf3q
2023/02/16 17:45:39 [ERROR] deleting RDS Instance Automated Backup (arn:aws:rds:us-west-2:123456789012:auto-backup:ab-dkedyvwfa2niqxut6p23q3yaz2jli2utyoobf3q): InvalidDBInstanceAutomatedBackupState: You can't delete automated backups.
	status code: 400, request id: 84ccbbd5-5534-46c5-af13-169875a9a7a7
2023/02/16 17:45:39 [DEBUG] Deleting RDS Instance Automated Backup: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-pxokasoxjknalh3xd6qqyve7zcqxz2m6txij7iq
2023/02/16 17:45:39 [ERROR] deleting RDS Instance Automated Backup (arn:aws:rds:us-west-2:123456789012:auto-backup:ab-pxokasoxjknalh3xd6qqyve7zcqxz2m6txij7iq): InvalidDBInstanceAutomatedBackupState: You can't delete automated backups.
	status code: 400, request id: e9894be5-ca3a-4185-b390-3e0235612f09
2023/02/16 17:45:39 [DEBUG] Deleting RDS Instance Automated Backup: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-rvdjvrrsiydqdespjwrlitkiqdf5yxhuoraqnky
2023/02/16 17:45:39 [ERROR] deleting RDS Instance Automated Backup (arn:aws:rds:us-west-2:123456789012:auto-backup:ab-rvdjvrrsiydqdespjwrlitkiqdf5yxhuoraqnky): InvalidDBInstanceAutomatedBackupState: You can't delete automated backups.
	status code: 400, request id: 5d5c3f11-8357-4dfc-8aab-c27088c01fe2
2023/02/16 17:45:39 [DEBUG] Deleting RDS Instance Automated Backup: arn:aws:rds:us-west-2:123456789012:auto-backup:ab-srqgk6fdu6hghmeciso2xfyzat2akho3aqu632y
2023/02/16 17:45:39 [ERROR] deleting RDS Instance Automated Backup (arn:aws:rds:us-west-2:123456789012:auto-backup:ab-srqgk6fdu6hghmeciso2xfyzat2akho3aqu632y): InvalidDBInstanceAutomatedBackupState: You can't delete automated backups.
	status code: 400, request id: fdfa5a0a-4233-4ae1-bf9a-7ed3b5a8359b
2023/02/16 17:45:39 [DEBUG] Completed Sweeper (aws_db_instance_automated_backups_replication) in region (us-west-2) in 1.908803167s
2023/02/16 17:45:39 [DEBUG] Sweeper (aws_neptune_cluster_instance) has dependency (aws_rds_global_cluster), running..
2023/02/16 17:45:39 [DEBUG] Sweeper (aws_rds_global_cluster) already ran in region (us-west-2)
2023/02/16 17:45:39 [DEBUG] Running Sweeper (aws_neptune_cluster_instance) in region (us-west-2)
2023/02/16 17:45:40 [DEBUG] Completed Sweeper (aws_neptune_cluster_instance) in region (us-west-2) in 54.151083ms
2023/02/16 17:45:40 [DEBUG] Sweeper (aws_neptune_cluster) has dependency (aws_neptune_cluster_instance), running..
2023/02/16 17:45:40 [DEBUG] Sweeper (aws_neptune_cluster_instance) has dependency (aws_rds_global_cluster), running..
2023/02/16 17:45:40 [DEBUG] Sweeper (aws_rds_global_cluster) already ran in region (us-west-2)
2023/02/16 17:45:40 [DEBUG] Sweeper (aws_neptune_cluster_instance) already ran in region (us-west-2)
2023/02/16 17:45:40 [DEBUG] Running Sweeper (aws_neptune_cluster) in region (us-west-2)
2023/02/16 17:45:40 [DEBUG] Completed Sweeper (aws_neptune_cluster) in region (us-west-2) in 42.231125ms
2023/02/16 17:45:40 Completed Sweepers for region (us-west-2) in 2.649851208s
2023/02/16 17:45:40 Sweeper Tests for region (us-west-2) ran successfully:
	- aws_rds_cluster_parameter_group
	- aws_db_instance_automated_backups_replication
	- aws_neptune_cluster_instance
	- aws_neptune_cluster
	- aws_rds_global_cluster
	- aws_db_instance
	- aws_rds_cluster
```

```console
$ make testacc PKG=rds TESTS=TestAccRDSGlobalCluster_

--- PASS: TestAccRDSGlobalCluster_disappears (21.91s)
--- PASS: TestAccRDSGlobalCluster_basic (27.25s)
--- PASS: TestAccRDSGlobalCluster_Engine_aurora (27.27s)
--- PASS: TestAccRDSGlobalCluster_EngineVersion_auroraPostgreSQL (27.40s)
--- PASS: TestAccRDSGlobalCluster_EngineVersion_auroraMySQL (27.51s)
--- PASS: TestAccRDSGlobalCluster_EngineVersion_aurora (27.64s)
--- PASS: TestAccRDSGlobalCluster_storageEncrypted (34.78s)
--- PASS: TestAccRDSGlobalCluster_databaseName (37.00s)
--- PASS: TestAccRDSGlobalCluster_deletionProtection (63.78s)
--- PASS: TestAccRDSGlobalCluster_sourceDBClusterIdentifier (124.87s)
--- PASS: TestAccRDSGlobalCluster_SourceDBClusterIdentifier_storageEncrypted (144.70s)
--- PASS: TestAccRDSGlobalCluster_EngineVersion_updateMinor (1713.59s)
--- PASS: TestAccRDSGlobalCluster_EngineVersion_updateMajor (2004.62s)
--- PASS: TestAccRDSGlobalCluster_EngineVersion_updateMinorMultiRegion (3318.77s)
--- PASS: TestAccRDSGlobalCluster_EngineVersion_updateMajorMultiRegion (6336.05s)
```

### CODE REVIEW

File: internal/service/neptune/sweep.go [hunk 1]
```diff
@@ -33,6 +34,9 @@ func init() {
 	resource.AddTestSweepers("aws_neptune_cluster_instance", &resource.Sweeper{
 		Name: "aws_neptune_cluster_instance",
 		F:    sweepClusterInstances,
+		Dependencies: []string{
+			"aws_rds_global_cluster",
```
Comment (ewbankkit) [inline internal/service/neptune/sweep.go:38]: Should this be `aws_neptune_global_cluster`?

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

**Comment by ewbankkit** [comment]
My guess is that `aws_docdb_cluster_instance` should follow the same pattern as I believe that it is just a thin veneer over RDS.

File: internal/service/neptune/sweep.go (hunk 1 above)
Comment (gdavison) [reply internal/service/neptune/sweep.go:38]: Surprisingly, no. It looks like a Neptune global cluster is implemented using an RDS Global Cluster. I'll add a comment to clarify that.

**Comment by gdavison** [comment]
> My guess is that `aws_docdb_cluster_instance` should follow the same pattern as I understand that it is just a thin veneer over RDS.

I'll update that, as well

**Pull request merged**